### PR TITLE
Adding the fix related to the issue #157

### PR DESCRIPTION
**Pull request by edgar-pek**
Added the check and the appropriate unit test.

### CODE REVIEW

File: src/google/protobuf/stubs/stringprintf.cc [hunk 1]
```diff
@@ -85,6 +85,12 @@ void StringAppendV(string* dst, const char* format, va_list ap) {
       // Just an error.
       return;
     }
+
+    if(result == std::numberic_limits<int>::max()){
```
Comment (xfxyjwf) [inline src/google/protobuf/stubs/stringprintf.cc]: Move this to before "int length = reuslt+1"?

File: src/google/protobuf/stubs/stringprintf_unittest.cc [hunk 2]
```diff
@@ -147,6 +147,16 @@ TEST(StringPrintfTest, LargeBuf) {
   delete[] buf;
 }
 
+TEST(StringPrintfTest, HugeBuf) {
+  // Check that the a corner case buffer is handled correctly.
+  int n = std::numeric_limits<int>::max() ;
+  char* buf = (char *) malloc(n);
```
Comment (xfxyjwf) [inline src/google/protobuf/stubs/stringprintf_unittest.cc]: Check whether buf is NULL before using it. We should make the test pass if the machine does not have 2G memory at all.

**Comment by xfxyjwf** [comment]
Please fix the build failures.

**Comment by xfxyjwf** [comment]
Closing due to inactivity.
